Source/WebCore/dom/Document.cpp

```diff
@@ -6630,7 +6630,14 @@ void Document::whenWindowLoadEventOrDestroyed(CompletionHandler<void()>&& comple
         completionHandler();
         return;
     }
-    m_whenWindowLoadEventOrDestroyed = WTFMove(completionHandler);
+    if (!m_whenWindowLoadEventOrDestroyed) {
+        m_whenWindowLoadEventOrDestroyed = WTFMove(completionHandler);
+        return;
+    }
+    m_whenWindowLoadEventOrDestroyed = [oldCompletionHandler = WTFMove(m_whenWindowLoadEventOrDestroyed), newCompletionHandler = WTFMove(completionHandler)]() mutable {
```

Comment:
```suggestion
    m_whenWindowLoadEventOrDestroyed = [oldCompletionHandler = std::exchange(m_whenWindowLoadEventOrDestroyed, { }), newCompletionHandler = WTFMove(completionHandler)]() mutable {
```

We do not tend to move member variables since they can be reused.